src/hooks/useSendEmail.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  Dispatch,
+  SetStateAction,
+  useCallback,
+  useContext,
+} from 'react';
+import { useRouter } from 'next/router';
+import { useSearchParams } from 'next/navigation';
+import { AppContext } from '@context';
+import { decodeText } from '@utils';
+import { emailService } from '@services';
+import { useToast } from '@chakra-ui/react';
+
+type EmailServiceProps = {
+  onClose: () => void;
+  responderName: string;
+  setLoading: Dispatch<SetStateAction<boolean>>;
+};
+
+export const useSendEmail = () => {
+  const searchParams = useSearchParams();
+  const { selectedAnswer } = useContext(AppContext);
+  const toast = useToast();
+  const router = useRouter();
+
+  const encodedSenderName = searchParams.get('sn')!;
+  const senderName = decodeText(encodedSenderName);
+  const senderEmail = searchParams.get('se')!;
+
+  const sendEmail = useCallback(
+    async ({
```

Comment:
It's not clear to me that all these params should be parameterized on the returned function, as opposed to parameterized at the hook call with the returned function acting as a closure. Generally, any value should be constant over the lifetime of the hook should be passed into the hook. Parameterize as close to the point of not-knowing as possible.